File: domain/src/main/java/no/nav/melosys/domain/dokument/inntekt/tillegsinfo/BarnepensjonOgUnderholdsbidrag.java
package no.nav.melosys.domain.dokument.inntekt.tillegsinfo;

import com.fasterxml.jackson.annotation.JsonIgnore;
import no.nav.melosys.domain.ErPeriode;
import no.nav.melosys.domain.HarPeriode;
import no.nav.melosys.domain.dokument.inntekt.Periode;

public class BarnepensjonOgUnderholdsbidrag extends TilleggsinformasjonDetaljer implements HarPeriode {

    public String forsørgersFødselnummer;

    public Periode tidsrom;

    @Override
    @JsonIgnore
    public ErPeriode getPeriode() {
        return tidsrom;
    }
}

File: domain/src/main/kotlin/no/nav/melosys/domain/dokument/inntekt/tillegsinfo/BarnepensjonOgUnderholdsbidrag.kt
package no.nav.melosys.domain.dokument.inntekt.tillegsinfo

import com.fasterxml.jackson.annotation.JsonIgnore
import no.nav.melosys.domain.ErPeriode
import no.nav.melosys.domain.HarPeriode
import no.nav.melosys.domain.dokument.inntekt.Periode
import no.nav.melosys.domain.toErPeriode

class BarnepensjonOgUnderholdsbidrag : TilleggsinformasjonDetaljer, HarPeriode {
    var forsørgersFødselnummer: String? = null
    var tidsrom: Periode? = null
    @JsonIgnore
    override fun getPeriode(): ErPeriode? {
        return tidsrom?.toErPeriode()
    }
}
